scipy/stats/stats.py

```diff
@@ -4259,6 +4265,529 @@ def weightedtau(x, y, rank=True, weigher=None, additive=True):
+    .. [2] Panda, S., Palaniappan, S., Xiong, J., Swaminathan, A.,
+           Ramachandran, S., Bridgeford, E. W., ... Vogelstein, J. T. (2019).
+           mgcpy: A Comprehensive High Dimensional Independence Testing Python
+           Package. ArXiv:1907.02088 [Cs, Stat].
+    .. [3] Shen, C., Priebe, C.E., & Vogelstein, J. T. (2019). From distance
+           correlation to multiscale graph correlation. Journal of the American
+           Statistical Association.
+    .. [4] Shen, C. & Vogelstein, J. T. (2018). The Exact Equivalence of
+           Distance and Kernel Methods for Hypothesis Testing. ArXiv:1806.05514
+           [Cs, Stat].
+
+    Examples
+    --------
+    >>> from scipy.stats import multiscale_graphcorr
+    >>> np.random.seed(12345678)
+    >>> x = np.linspace(-1, 1, num=100)
+    >>> y = x + 0.3 * np.random.random(x.size)
+    >>> stat, pvalue, _ = multiscale_graphcorr(x, y)
+    >>> round(stat, 1), round(pvalue, 1)
+    (1.0, 0.0)
+
+    Alternatively,
+
+    >>> from scipy.stats import multiscale_graphcorr
+    >>> np.random.seed(12345678)
+    >>> x = np.linspace(-1, 1, num=100)
+    >>> y = x + 0.3 * np.random.random(x.size)
+    >>> mgc = multiscale_graphcorr(x, y)
+    >>> round(mgc.stat, 1), round(mgc.pvalue, 1)
+    (1.0, 0.0)
+
+    To run an unpaired two-sample test,
+
+    >>> from scipy.stats import multiscale_graphcorr
+    >>> np.random.seed(12345678)
+    >>> x = np.random.binomial(100, 0.5, size=(100, 5))
+    >>> y = np.random.normal(0, 1, size=(80, 5))
+    >>> mgc = multiscale_graphcorr(x, y)
+    >>> round(mgc.stat, 1), round(mgc.pvalue, 1)
+    (1.0, 0.0)
+
+    or, if shape of the inputs are the same,
+
+    >>> from scipy.stats import multiscale_graphcorr
+    >>> np.random.seed(12345678)
+    >>> x = np.random.binomial(100, 0.5, size=(100, 5))
+    >>> y = np.random.normal(0, 1, size=(100, 5))
+    >>> mgc = multiscale_graphcorr(x, y, is_twosamp=True)
+    >>> round(mgc.stat, 1), round(mgc.pvalue, 1)
+    (1.0, 0.0)
+    """
+    if not isinstance(x, np.ndarray) or not isinstance(y, np.ndarray):
+        raise ValueError("x and y must be ndarrays")
+
+    # convert arrays of type (n,) to (n, 1)
+    if x.ndim == 1:
+        x.shape = (-1, 1)
+    if y.ndim == 1:
+        y.shape = (-1, 1)
+
+    nx, px = x.shape
+    ny, py = y.shape
+
+    # check for NaNs
+    _contains_nan(x, nan_policy='raise')
+    _contains_nan(y, nan_policy='raise')
+
+    if nx != ny:
+        if px == py:
+            # reshape x and y for two sample testing
+            is_twosamp = True
+        else:
+            raise ValueError("Shape mismatch, x and y must have shape [n, p] "
+                             "and [n, q] or have shape [n, p] and [m, p].")
+
+    if nx < 5 or ny < 5:
+        raise ValueError("MGC requires at least 5 samples to give reasonable "
+                         "results.")
+
+    # convert x and y to float
+    x = np.asarray(x).astype(np.float64)
```

Comment:
you already checked that `x` and `y` were `ndarray` instances, so `asarray` can be removed

Reply:
https://github.com/scipy/scipy/pull/10524/commits/fabfbb13356c806099a19d3ce7ed5448ba2219d0